### ISSUE TITLE
misunderstanding of the argument idx.incomplete in produce_NA (How to simulate missings)

### ISSUE THREAD

**Issue by rabea-a** [R-miss-tastic/website#53]
Hello @imkemayer ,
thank you for your quick and detailed answer in the my pull request #52 .
Sorry for having problems with the argument idx.incomplete! Even with the new commit the code leads to an error for my data. Maybe the following example will clearify my problem:

In your answer you mentioned the argument to be "a binary array of size p=#variables, an entry=1 meaning that the corresponding variable will have missing values". So below you can see my minimum example code, I tried to specify the argument with an array.

```{r} 
x1 <- as.factor(sample(c("A","B"), 20, replace = TRUE, prob = c(0.75, 0.25)))
x2 <- rnorm(20)
x <- data.frame(x1,x2)

data_NA <- produce_NA(data = x, mechanism = "MAR", perc.missing = 0.2, 
                      idx.incomplete = array(c(1, 0)), idx.covariates = c(0, 1))
# missings were generated for variable x1

data_NA <- produce_NA(data = x, mechanism = "MAR", perc.missing = 0.2, 
                      idx.incomplete = array(c(0,1)), idx.covariates = c(1,0))
# error
``` 

Can you help me fix this error?

Gratefully,
Rabea

**Comment by imkemayer** [comment]
Hi @rabea-a,

Thanks again for reaching out, every comment and counterexample helps us improve our workflows, so it's great that you've looked so closely at our `produce_NA` function and pointed out the above problem.

Indeed, there was still a bug in the function related to the `idx.incomplete` argument (we were multiple contributors to this function and due to the lack of documentation for the function arguments, there seems to have been different understandings on how this specific argument was defined). Hopefully the last commit fixed the issue (4f80cde5e860ae19cbe50c58b45b068f8fa319f8). At least I think, the example you provided is no running correctly.

I will do some more checks later on to check that there are no more bugs related to `idx.incomplete`, but if you find some problems, your feedback is very welcome!

Best
Imke

**Comment by rabea-a** [comment]
Hi @imkemayer,
thanks again for your quick reply! Unfortunately I don't have much time today, but it looks like it's going to work now. Thank you!

Yes, I have also seen the different view of the argument idx.complete in the code (but I think the wrong way around). Maybe you can adjust the explanation of the parameter more precisely. And you should also adjust the example code on page 10 of the PDF [How to generate missing values](https://rmisstastic.netlify.app/how-to/generate/missSimul.pdf). This was leading me into the wrong direction.

Have a nice weekend
Rabea
